docs/concepts/advanced/.nav.yml

```diff
@@ -6,9 +6,9 @@ nav:
   - Lifecycle hooks: hooks.md
   - Registering components: component_registry.md
   - Component caching: component_caching.md
-  - Typing and validation: typing_and_validation.md
+  - Component context and scope: component_context_scope.md
```

Comment:
I've moved the page on component context to the "advanced" section, and the typing to the "fundamentals".